docs/adr/000x-modelling-history.md

```diff
@@ -0,0 +1,314 @@
+represented below:
+
+![Image Revision breakdown](000x/image-revision-diagram.png)
+
+The Image model itself is used for continuation between image revisions, we know
+that two Image::Revisions are versions of the same item if they share the same
+Image association.
+
+The data of an Image::Revision is stored between an Image::FileRevision and an
+Image::MetadataRevision. These two objects differ by the fact that any change to
+Image::FileRevision requires changes to the resultant files (such as crop
+dimensions), whereas Image::MetadataRevision stores accompanying data that
+doesn't affect the files (such as alt text).
+
+Image::FileRevision is associated with an ActiveStorage::Blob object that is
+responsible for managing the storage of the source file. It also has a one to
+many association with Image::Asset, each of these represent resulting files that
+are uploaded to Asset Manager. For a single image uploaded we'd likely upload
+multiple sizes to Asset Manager - these different files are named variants. An
+Image::Asset object stores whether the asset is on Asset Manager, what state it
+is in and the URL.
+
+### Timeline entry
+
+A timeline entry represents an event that should be shown to a user as part
+of a documents history. This stores the event which occurred and associations
+to relevant models. The sole purpose of this is for representing a
+timeline to users visually and not for storing historical metadata.
+
+The TimelineEntry model should not store data which could not be
+derived from other aspects of a document. This is to maintain
+flexibility for timeline to be an evolving and iterated part of the
+application.
+
+At the time of writing it wasn't yet determined what the
+timeline would show, and therefore it wasn't clear exactly how
+best to model them. Because of this TimelineEntry will be modelled in a
+speculative way with a number of joins to relevant data.
+
+### Topics / Taxonomy
+
+No data related to topics / taxonomy is intended to be stored in Content
+Publisher at the current point in time. This is due to this being editable
+from other locations, most notably Content Tagger, and therefore difficult for
+us to store an accurate picture. As we learn more this may be reconsidered.
+
+## Status
+
+Accepted
+
+## Consequences
+
+This modelling should provide a basis to provide the richer history
+functionality that can enable meeting and exceeding the features offered by
+Whitehall and Mainstream Publisher.
+
+It does however come with a number of concerns that are highlighted below.
+
+### Mo' models, mo' problems
+
+With an increase in the number of models there is more for a developer to
+understand and a higher barrier for entry. For every action there are either
+more database queries or a more complicated single query to look up information.
+
+Some ideas to alleviate this are:
+
+- the use of [delegate][] with ActiveRecord to make it simpler to access data
+  without interacting with individual models;
```

Comment:
A note on this - I agree with the use of delegate but can be quite confusing when it's nested through multiple models

Reply:
Yup absolutely, they're definitely a trade off and with each level they get harder to trace to origin.